### PR TITLE
fix: Wrong SPF for CA

### PR DESCRIPTION
**Pull request by mohammed-zahaf**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | DTRSD-25117
| License          | BSD 3-Clause

## Description

Aims of this fix is to set the SPF value to CA or COM value.

### CODE REVIEW

**Review by Heiiji — APPROVED** [review]

File: packages/manager/apps/web/client/app/domain/zone/record/domain-zone-record.controller.js [hunk 1]
```diff
@@ -8,6 +8,8 @@ import isEmpty from 'lodash/isEmpty';
 import isString from 'lodash/isString';
 import trim from 'lodash/trim';
 import set from 'lodash/set';
+import { Environment } from '@ovh-ux/manager-config';
+import { CANADA_REGION } from '../../../../../../../modules/config/src/environment/environment.constants';
```
Comment (cbourgois) [inline packages/manager/apps/web/client/app/domain/zone/record/domain-zone-record.controller.js:12]: Why do you need to import `environment.constants` like this ?
If you really need it, constants should be exported by `@ovh-ux/manager-config` ?

Here `import ... from '@ovh-ux/manager-config';` and `import ... from '../../../../../../../modules/config/src/` target the same package.

File: packages/manager/apps/web/client/app/domain/zone/record/domain-zone-record.controller.js [hunk 2]
```diff
@@ -700,7 +702,9 @@ angular.module('App').controller(
 
     useSpfOvh() {
       this.model.target = {};
-      this.model.target.include = 'mx.ovh.com';
+      this.model.target.include = `mx.ovh.${
+        Environment.getRegion() === CANADA_REGION ? 'ca' : 'com'
+      }`;
```
Comment (cbourgois) [inline packages/manager/apps/web/client/app/domain/zone/record/domain-zone-record.controller.js:707]:
```suggestion
      this.model.target.include = Environment.getRegion() === 'CA' ? 'mx.ovh.ca' : 'mx.ovh.com';
```

As it seems to be an hotfix, I think we can get rid of refactor everything to use a constant and use `'CA'`.
The refacto should be done in another PR.

File: packages/manager/modules/config/src/environment/environment.constants.js [hunk 3]
```diff
@@ -1,5 +1,6 @@
 export const ALLOWED_REGIONS = ['EU', 'CA', 'US'];
 export const DEFAULT_REGION = 'EU';
+export const CANADA_REGION = 'CA';
```
Comment (cbourgois) [inline packages/manager/modules/config/src/environment/environment.constants.js:3]: The constant `CANADA_REGION` is exported but never used in `@ovh-ux/manager-config` package, and it's never exported in the dist.

We don't have `EUROPE_REGION` or `UNITED_STATES_REGION`. If we add them, we should at least export them in the dist (as `REGIONS:{EU, CA, US}` , and refactor every place in our code where `EU` / `CA` / `US` strings are used.

**Review by cbourgois — CHANGES_REQUESTED** [review]

File: packages/manager/apps/web/client/app/domain/zone/record/domain-zone-record.controller.js (hunk 1 above)
Comment (mohammed-zahaf) [reply packages/manager/apps/web/client/app/domain/zone/record/domain-zone-record.controller.js:12]: I had try create a static method `isCanadaRegion()`.
When i try to call it like so `Environment.isCanadaRegion()` i got error.
Now i understand the mistake the `Environment` is already an instance object not a Class.

So i don't need anymore the `CANADA_REGION` import.

File: packages/manager/apps/web/client/app/domain/zone/record/domain-zone-record.controller.js [hunk 4]
```diff
@@ -768,5 +769,13 @@ angular.module('App').controller(
...
         });
     }
+
+    static getSPFIncludeName() {
+      return `mx.ovh.${
+        Environment.constructor.isCanadaRegion(Environment.getRegion())
+          ? 'ca'
+          : 'com'
+      }`;
+    }
```
Comment (frenautvh) [inline packages/manager/apps/web/client/app/domain/zone/record/domain-zone-record.controller.js:779]:
```js
return `mx.ovh.${Environment.getRegion() === 'ca' ? 'ca' : 'com'}`
```
just do that, no need to change Environement

Comment (mohammed-zahaf) [reply packages/manager/apps/web/client/app/domain/zone/record/domain-zone-record.controller.js:775]: Ok changed

File: packages/manager/apps/web/client/app/domain/zone/record/domain-zone-record.controller.js (hunk 2 above)
Comment (mohammed-zahaf) [reply packages/manager/apps/web/client/app/domain/zone/record/domain-zone-record.controller.js]: Ok got it.

**Review by frenautvh — APPROVED** [review]

**Review by antleblanc — APPROVED** [review]
